clang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -1980,9 +1980,17 @@ ModuleLoadResult CompilerInstance::findOrCompileModuleAndReadAST(
 
     // Check whether M refers to the file in the prebuilt module path.
     if (M && M->getASTFile())
-      if (auto ModuleFile = FileMgr->getFile(ModuleFilename))
+      if (auto ModuleFile = expectedToOptional(FileMgr->getFileRef(ModuleFilename))) {
         if (*ModuleFile == M->getASTFile())
           return M;
+#if !defined(__APPLE__)
+        // Workaround for ext4 file system. Also check bypass file if exists.
+        if (auto Bypass = FileMgr->getBypassFile(*ModuleFile,
```

Comment:
Just wondering: would changing `getBypassFile` to use `getNameAsRequested()` work instead of passing in an explicit name?  Or does that break something else?

Reply:
This is to minimize the downstream change. I didn't test the alternative because it requires passing a `FileEntryRef` into the function, rather than a `FileEntry`.

Reply:
Wait, it has always been `FileEntryRef`. I probably misread the function. Let me try that.